test/files/jvm/scala-concurrent-tck.scala

```diff
@@ -989,36 +986,34 @@ class CustomExecutionContext extends TestBase {
     assert(count >= 1)
   }
 
-  def testUncaughtExceptionReporting(): Unit = once {
-    done =>
-      import java.util.concurrent.TimeUnit.SECONDS
-      val example = new InterruptedException()
-      val latch = new java.util.concurrent.CountDownLatch(1)
-      @volatile var thread: Thread = null
-      @volatile var reported: Throwable = null
-      val ec = ExecutionContext.fromExecutorService(null, t => {
-        reported = t
-        latch.countDown()
-      })
+  def testUncaughtExceptionReporting(): Unit = once { done =>
+    val example = new InterruptedException
+    val latch = new CountDownLatch(1)
+    @volatile var thread: Thread = null
+    @volatile var reported: Throwable = null
+    val ec = ExecutionContext.fromExecutorService(null, t => {
+      reported = t
+      latch.countDown()
+    })
 
-      @tailrec def waitForThreadDeath(turns: Int): Boolean =
-          if (turns <= 0) false
-          else if ((thread ne null) && thread.isAlive == false) true
-          else {
-            Thread.sleep(10)
-            waitForThreadDeath(turns - 1)
-          }
+    @tailrec def waitForThreadDeath(turns: Int): Boolean =
+      turns > 0 && (thread != null && !thread.isAlive || { Thread.sleep(10L) ; waitForThreadDeath(turns - 1) })
```

Comment:
why are we testing `Thread#isAlive` and not `Thread#isInterrupted`, or at least testing both? at least from my reading of it, [`isAlive`](https://download.java.net/java/early_access/jdk17/docs/api/java.base/java/lang/Thread.html#isAlive()) tests if the thread has fallen victim to `ThreadDeath`, which is only thrown by deprecated methods on `Thread` (such as `Thread#stop`). If a thread is a member of a pool, there's no particular reason it ought to die unnecessarily.

Reply:
I don't know yet why the thread used to terminate. The ueh was always set, so something else is going on. I might have to up my Klang level.

The test wants to ensure that the exception is reported exactly once; it could also do that by waiting for pool to shutdown instead of state of a particular thread.